src/hooks/cache/caching.js

```diff
@@ -0,0 +1,61 @@
+import { useReducer } from 'react';
+import { setCaching } from './cacheActions';
+
+function reducer(state, action) {
+	switch (action.set_cache.type) {
+		case 'SET_CACHE':
+			if (Array.isArray(action.key) && action.key.length > 1) {
+				var cidCache = action.key.reduce((ac, cid) => {
+					return {
+						...ac,
+						[cid]: action.payload,
+					};
+				}, {});
+				var purlCache = action.payload[0].prettyUrlName
```

Comment:
Why is there any reference to `purl` or `prettyUrlName` in this file?

Reply:
my bad, forgot to update this part of the code, regarding iterating through the c-codes